src/lib/ThumbnailsSidebar.js

```diff
@@ -201,6 +206,14 @@ class ThumbnailsSidebar {
             thumbnailEl.classList.add(CLASS_BOX_PREVIEW_THUMBNAIL_IS_SELECTED);
         }
 
+        // If image is already in cache, then use it instead of waiting for
+        // the second render image pass
+        const cachedImage = this.thumbnailImageCache.get(itemIndex);
+        if (cachedImage && !cachedImage.inProgress) {
```

Comment:
if It's in progress, that means there is no generated image yet so there's nothing to insert into the DOM at this point. We add a cache entry with `inProgress` as true to avoid multiple images being requested